corehq/apps/geospatial/tasks.py

```diff
@@ -14,3 +29,47 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@task(queue='geospatial_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, ES_INDEX_TASK_HELPER_BASE_KEY)
+    geo_case_prop = get_geo_case_property(domain)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.mark_as_error(error_slug='TOO_MANY_CASES')
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = math.ceil(doc_count / DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            docs_left = doc_count - (DEFAULT_QUERY_LIMIT * i)
+            limit = min(DEFAULT_QUERY_LIMIT, docs_left)
+            process_batch(
+                domain,
+                geo_case_prop,
+                case_type=None,
+                query_limit=limit,
+                chunk_size=DEFAULT_CHUNK_SIZE,
+                offset=i * DEFAULT_QUERY_LIMIT,
+                total_count=doc_count
+            )
+            current_batch = i + 1
+            celery_task_tracker.update_progress(
+                current=current_batch * min(DEFAULT_CHUNK_SIZE, doc_count),
```

Comment:
should this be `DEFAULT_QUERY_LIMIT` and not `DEFAULT_CHUNK_SIZE`?
since each batch is `DEFAULT_QUERY_LIMIT` big?

Reply:
You are completely right, this should be `DEFAULT_QUERY_LIMIT`. Will correct this.